server/Arcadia.Assistant.Calendar.Notifications/SickLeaveAccountingEmailNotificationActor.cs

```diff
@@ -17,55 +17,79 @@
 
     public class SickLeaveAccountingEmailNotificationActor : UntypedActor
     {
-        private readonly IEmailWithFixedRecipientNotification approvedEmailNotificationConfig;
+        private readonly IEmailWithFixedRecipientNotification createdEmailNotificationConfig;
         private readonly IEmailWithFixedRecipientNotification prolongedEmailNotificationConfig;
+        private readonly IEmailWithFixedRecipientNotification cancelledEmailNotificationConfig;
         private readonly IActorRef organizationActor;
 
         private readonly ILoggingAdapter logger = Context.GetLogger();
 
         public SickLeaveAccountingEmailNotificationActor(
-            IEmailWithFixedRecipientNotification approvedEmailNotificationConfig,
+            IEmailWithFixedRecipientNotification createdEmailNotificationConfig,
             IEmailWithFixedRecipientNotification prolongedEmailNotificationConfig,
+            IEmailWithFixedRecipientNotification cancelledEmailNotificationConfig,
             IActorRef organizationActor)
         {
-            this.approvedEmailNotificationConfig = approvedEmailNotificationConfig;
+            this.createdEmailNotificationConfig = createdEmailNotificationConfig;
             this.prolongedEmailNotificationConfig = prolongedEmailNotificationConfig;
+            this.cancelledEmailNotificationConfig = cancelledEmailNotificationConfig;
             this.organizationActor = organizationActor;
 
+            Context.System.EventStream.Subscribe<CalendarEventCreated>(this.Self);
             Context.System.EventStream.Subscribe<CalendarEventChanged>(this.Self);
         }
 
         protected override void OnReceive(object message)
         {
             switch (message)
             {
-                case CalendarEventChanged msg
-                    when msg.NewEvent.Type == CalendarEventTypes.Sickleave &&
-                    msg.NewEvent.Status == SickLeaveStatuses.Approved:
+                case CalendarEventCreated msg when
+                    msg.Event.Type == CalendarEventTypes.Sickleave:
 
-                    var isProlonged = msg.OldEvent.Status == SickLeaveStatuses.Approved;
-                    var dateChanged = msg.OldEvent.Dates.EndDate != msg.NewEvent.Dates.EndDate;
+                    this.organizationActor
+                        .Ask<EmployeesQuery.Response>(EmployeesQuery.Create().WithId(msg.Event.EmployeeId))
+                        .ContinueWith(task => new CalendarEventChangedWithAdditionalData(msg.Event, NotificationType.Created, task.Result.Employees.FirstOrDefault()?.Metadata))
+                        .PipeTo(this.Self);
+                    break;
 
-                    if (isProlonged && !dateChanged)
-                    {
-                        return;
-                    }
+                case CalendarEventChanged msg when
+                    msg.NewEvent.Type == CalendarEventTypes.Sickleave &&
+                    msg.NewEvent.Status == SickLeaveStatuses.Approved &&
```

Comment:
SickLeaveStatuses should not contain Approved anymore

Reply:
Ok then, what event statuses flow should be in this case?
It was Requested -> Approved (by manager) | Cancelled (by employee) -> Rejected (by manager) | Completed (by accounting).
After my changes it transformed to Requested -> Approved (automatically) | Cancelled (by employee) -> Cancelled (by employee) | Rejected (not used) | Completed (by accounting).

Reply:
Requested (can be prolonged) -> Cancelled (by employee) / Completed (by accounting). You may rename Requested to 'Not Completed' if requested/approved confuse you.

Reply:
Automatic approval is weird.

Reply:
Done.